parser/src/ast.rs

```diff
@@ -137,7 +137,8 @@ pub enum Expression {
     },
     Call {
         function: Box<Expression>,
-        args: Vec<Expression>,
+        // parameters are (None, value), kwargs are (keyword name, value)
+        args: Vec<(Option<String>, Expression)>,
```

Comment:
Are kwargs allowed before normal args? If not, I would suggest two vectors, args and kwargs. In any case I would like to have a seperate struct for kwargs with a name and expression member.

Reply:
They aren't, but these are created in the lalrpop-generated code. I couldn't work out a good way of expressing the ordering there, so at the moment, we enforce the ordering in the compiler.

The problem is, I think, that with `foo(x, y="z")` the parser can't know when it examines `y` whether or not it is a positional argument or a keyword argument.

My grammar-fu is very weak though, so there may well be a way of handling that!

Reply:
Okay, I'll have a look later on.